codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -159,11 +159,15 @@ public boolean goAwayReceived() {
 
     @Override
     public void goAwayReceived(int lastKnownStream, long errorCode, ByteBuf debugData) {
-        boolean alreadyNotified = goAwayReceived();
         localEndpoint.lastKnownStream(lastKnownStream);
-        if (!alreadyNotified) {
-            for (Listener listener : listeners) {
-                listener.onGoAwayReceived(lastKnownStream, errorCode, debugData);
+        for (Listener listener : listeners) {
+            listener.onGoAwayReceived(lastKnownStream, errorCode, debugData);
+        }
+
+        Http2Stream[] streams = new Http2Stream[numActiveStreams()];
+        for (Http2Stream stream : activeStreams().toArray(streams)) {
+            if (stream.id() > lastKnownStream && localEndpoint.createdStreamId(stream.id())) {
```

Comment:
`stream.id()` may be used multiple times...should we put it in a temporary variable?